### DISCUSSION TITLE
Package for retrieval of model output

### DISCUSSION THREAD

**Discussion by pmav99** [oceanmodeling/Management#23] [Ideas]
@saeed-moghimi-noaa @SorooshMani-NOAA @brey @tomsail

In continuation of the discussion we had earlier today, some initial thoughts.

So, it should be possible to create a new package whose scope would be similar to that of `searvey`, but the primary focus would be on the retrieval of metadata and output of the supported models (STOFS-2D/3D etc).

The potential metadata for each supported model could include the period of available data, locations of station output, coverage mask, coastlines, etc.

Under the aspect of data retrieval, there are several features that could be incorporated:

1. A unified API for retrieving raw data from S3/Blob/Whatever buckets, eliminating the need to know the specific details such as the naming scheme of each model.
2. An optional feature for “normalizing” the model output. This would involve renaming dimensions/variables, etc., to a common “schema” to facilitate easier comparisons between the models. While this is already being done for STOFS-3D where renaming is done to follow the ADCIRC conventions, given the broader scope (e.g., support for MPAS or any other relevant model) it might be beneficial to avoid model-specific schemas.
3. Potential integration with the work being done on subsetting data and/or kerchunk deployments.

I am afraid I don't know the github handle of Moji. Feel free to ping her or whomever else is interested.

**Comment by SorooshMani-NOAA** [comment]
This is also a relevant topic in the code sprint: https://github.com/ioos/ioos-code-sprint/issues/44
